src/pages/iou/steps/IOUAmountPage/index.js

```diff
@@ -0,0 +1,179 @@
+        this.state = {
+            amount: props.selectedAmount,
+        };
+    }
+
+    componentDidMount() {
+        // Setup and attach keypress handler for navigating to the next screen
+        if (document) {
+            this.keyDownHandler = (keyBoardEvent) => {
+                // Proceeds to the next screen if the amount isn't empty
+                if (keyBoardEvent.key === 'Enter' && this.state.amount !== '') {
+                    this.props.onStepComplete(this.state.amount);
+                }
+            };
+            document.addEventListener('keydown', this.keyDownHandler);
+        }
+
+        // Input doesn't exist yet, likely due to navigation libraries delaying mount
+        // Wait until interactions are complete before trying to focus input
+        InteractionManager.runAfterInteractions(() => {
+            if (this.textInput) {
+                this.textInput.focus();
+            }
+        });
+    }
+
+    componentWillUnmount() {
+        // Cleanup all keydown event listeners that we've set up
+        if (document) {
+            document.removeEventListener('keydown', this.keyDownHandler);
+        }
+    }
+
+    /**
+     * Update amount with number or Backspace pressed.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
+     *
+     * @param {String} key
+     */
+    updateAmountIfValidInput(key) {
+        // Backspace button is pressed
+        if (key === '<' || key === 'Backspace') {
+            if (this.state.amount.length > 0) {
+                this.setState(prevState => ({
+                    amount: prevState.amount.substring(0, prevState.amount.length - 1),
+                }));
+            }
+            return;
+        }
+
+        this.setState((prevState) => {
+            const newValue = `${prevState.amount}${key}`;
+
+            // Regex to validate decimal number with up to 3 decimal numbers
+            const decimalNumberRegex = new RegExp(/^\d+(\.\d{0,3})?$/, 'i');
+            if (!decimalNumberRegex.test(newValue)) {
+                return prevState;
+            }
+            return {
+                amount: newValue,
+            };
+        });
+    }
+
+    render() {
+        return (
+            <View style={[styles.flex1, styles.pageWrapper]}>
+                <View style={[
+                    styles.flex1,
+                    styles.flexRow,
+                    styles.w100,
+                    styles.alignItemsCenter,
+                    styles.justifyContentCenter,
+                ]}
+                >
+                    <Text style={styles.iouAmountText}>
+                        {this.props.selectedCurrency}
+                    </Text>
+                    {this.props.isSmallScreenWidth
+                        ? <Text style={styles.iouAmountText}>{this.state.amount}</Text>
+                        : (
+                            <TextInputAutoWidth
```

Comment:
Why isn't the `autoFocus` property being used? I think that's specifically what it's for.